src/bot.ts

```diff
@@ -513,79 +571,23 @@ export class DiscordBot {
 
     // Update presence because sometimes discord misses people.
-    this.userSync.OnUpdateUser(msg.author).then(() => {
-      return this.GetRoomIdsFromChannel(msg.channel).catch((err) => {
-        log.verbose("No bridged rooms to send message to. Oh well.");
-        return null;
-      });
-    }).then((rooms) => {
-      if (rooms === null) {
-        return null;
-      }
-      const intent = this.GetIntentFromDiscordMember(msg.author);
-      // Check Attachements
-      msg.attachments.forEach((attachment) => {
-        Util.UploadContentFromUrl(attachment.url, intent, attachment.filename).then((content) => {
-          const fileMime = mime.lookup(attachment.filename);
-          const msgtype = attachment.height ? "m.image" : "m.file";
-          const info = {
-            mimetype: fileMime,
-            size: attachment.filesize,
-            w: null,
-            h: null,
-          };
-          if (msgtype === "m.image") {
-            info.w = attachment.width;
-            info.h = attachment.height;
-          }
-          rooms.forEach((room) => {
-            intent.sendMessage(room, {
-              body: attachment.filename,
-              info,
-              msgtype,
-              url: content.mxcUrl,
-              external_url: attachment.url,
-            }).then((res) => {
-              const evt = new DbEvent();
-              evt.MatrixId = res.event_id + ";" + room;
-              evt.DiscordId = msg.id;
-              evt.ChannelId = msg.channel.id;
-              evt.GuildId = msg.guild.id;
-              return this.store.Insert(evt);
-            });
-          });
-        });
-      });
-      if (msg.content !== null && msg.content !== "") {
-        this.msgProcessor.FormatDiscordMessage(msg).then((result) => {
-            rooms.forEach((room) => {
-              const trySend = () => intent.sendMessage(room, {
-                body: result.body,
-                msgtype: "m.text",
-                formatted_body: result.formattedBody,
-                format: "org.matrix.custom.html",
-              });
-              const afterSend = (res) => {
-                const evt = new DbEvent();
-                evt.MatrixId = res.event_id + ";" + room;
-                evt.DiscordId = msg.id;
-                evt.ChannelId = msg.channel.id;
-                evt.GuildId = msg.guild.id;
-                return this.store.Insert(evt);
-              };
-              trySend().then(afterSend).catch((e) => {
-                if (e.errcode !== "M_FORBIDDEN") {
-                  log.error("DiscordBot", "Failed to send message into room.", e);
-                  return;
-                }
-                return this.userSync.EnsureJoin(msg.member, room).then(() => trySend()).then(afterSend);
-              });
-            });
-        });
-      }
-    }).catch((err) => {
-      log.verbose("Failed to send message into room.", err);
-    });
+    await this.userSync.OnUpdateUser(msg.author);
+    let rooms;
```

Comment:
set default of null else it'd be undefined?
```suggestion
    let rooms = null;
```